### ISSUE TITLE
Remove minimum size rules for dynamic keyboards

### ISSUE THREAD

**Issue by kmcnaught** [OptiKey/OptiKey#785]
Currently you can't make the keyboard smaller than a certain minimum. This isn't necessary for dynamic keyboards, but some restriction must remain if/when sizes are propagated back to the settings

> 
> JuliusSweetland commented 6 days ago
> 
> @kmcnaught No objection to removing the restriction on dynamic keyboards, but I think there should be a safety mechanism on saving small heights so that it is impossible to accidentally set the height to, say, < 25 pixels.

> lsaranto commented 5 days ago
> 
> @kmcnaught @JuliusSweetland I'd be happy with anything between 1-5% of the screen size.

**Comment by AdamRoden** [comment]
@kmcnaught The real risk is not simply that a keyboard becomes too small, but that the keys on that keyboard become too small to activate. A keyboard with 10 columns that is 100 pixels wide would result in each key being only 10 pixels wide.

**Comment by kmcnaught** [comment]
Good point. I think the proposal is that for sizing of dynamic keyboards, we can let users shoot themselves in the foot (since dynamic keyboards are for more advanced users who should be able to get themselves out of that situation) but that we keep some limits on what sizes are persisted to the main optikey window settings. In that case at least we know how many rows/columns we might be dealing with as the builtin keyboards are fixed.

It could be that the restriction is merely moved to prevent PersistNewState having an effect if the resulting size is too low (but allowing smaller temporary states). In the case where someone tries to persist a tiny size, a warning toast could say they won't be persisted (which would be avoided by not persisting the state via the XML)

**Comment by lsaranto** [comment]
The most dangerous scenarios leading to too small keys are:
1. When making a new keyboard and e.g. making a typo in the config.
2. Resizing the keyboard window with the mouse or with the OptiKey actions.
3. Loading a random keyboard file without checking it first.

Did I miss anything? the number 2 is the one IMO that actually needs guards to prevent it from happening. As @kmcnaught mentioned the dynamic keyboards are an advanced feature.

This just came to my mind: What if there just was a new setting that drops the current (or revised) size restrictions? Either in the settings or in the dynamic keyboard XML file. A boolean that defaults to false. Then the user is actively selecting to ignore safety measures.

About the problem of persisting a too small size: When exactly would that come into play? I think I have not set it to true ever. What is the intended use case for the setting?
